pkg/resources/storage_integration.go

```diff
@@ -118,284 +120,262 @@ func StorageIntegration() *schema.Resource {
 	}
 }
 
-// CreateStorageIntegration implements schema.CreateFunc.
-func CreateStorageIntegration(d *schema.ResourceData, meta interface{}) error {
+func CreateStorageIntegration(d *schema.ResourceData, meta any) error {
 	db := meta.(*sql.DB)
-	name := d.Get("name").(string)
-
-	stmt := snowflake.NewStorageIntegrationBuilder(name).Create()
-
-	// Set required fields
-	stmt.SetString(`TYPE`, d.Get("type").(string))
-	stmt.SetBool(`ENABLED`, d.Get("enabled").(bool))
+	ctx := context.Background()
+	client := sdk.NewClientFromDB(db)
+
+	name := sdk.NewAccountObjectIdentifierFromFullyQualifiedName(d.Get("name").(string))
+	enabled := d.Get("enabled").(bool)
+	stringStorageAllowedLocations := expandStringList(d.Get("storage_allowed_locations").([]any))
+	storageAllowedLocations := make([]sdk.StorageLocation, len(stringStorageAllowedLocations))
+	for i, loc := range stringStorageAllowedLocations {
+		storageAllowedLocations[i] = sdk.StorageLocation{
+			Path: loc,
+		}
+	}
 
-	stmt.SetStringList("STORAGE_ALLOWED_LOCATIONS", expandStringList(d.Get("storage_allowed_locations").([]interface{})))
+	req := sdk.NewCreateStorageIntegrationRequest(name, enabled, storageAllowedLocations)
 
-	// Set optional fields
 	if v, ok := d.GetOk("comment"); ok {
-		stmt.SetString(`COMMENT`, v.(string))
+		req.WithComment(sdk.String(v.(string)))
 	}
 
 	if _, ok := d.GetOk("storage_blocked_locations"); ok {
-		stmt.SetStringList("STORAGE_BLOCKED_LOCATIONS", expandStringList(d.Get("storage_blocked_locations").([]interface{})))
+		stringStorageBlockedLocations := expandStringList(d.Get("storage_blocked_locations").([]any))
+		storageBlockedLocations := make([]sdk.StorageLocation, len(stringStorageBlockedLocations))
+		for i, loc := range stringStorageBlockedLocations {
+			storageBlockedLocations[i] = sdk.StorageLocation{
+				Path: loc,
+			}
+		}
 	}
 
-	if _, ok := d.GetOk("storage_aws_object_acl"); ok {
-		stmt.SetString("STORAGE_AWS_OBJECT_ACL", d.Get("storage_aws_object_acl").(string))
-	}
+	storageProvider := d.Get("storage_provider").(string)
 
-	// Now, set the storage provider
-	if err := setStorageProviderSettings(d, stmt); err != nil {
-		return err
+	switch storageProvider {
+	case "S3", "S3GOV":
```

Comment:
"S3", "S3gov", "GCS", "AZURE", "S3GOV" -> these are the values available, so specifying `S3gov` with such switch will result with an error return from the default branch.